core/store/src/trie/update.rs

```diff
@@ -39,6 +39,10 @@ pub struct TrieUpdate {
     prospective: TrieUpdates,
 }
 
+const fn assert_send_sync<T: Send + Sync>() {}
+#[allow(dead_code)]
+const _: () = assert_send_sync::<TrieUpdate>();
+
```

Comment:
Do you mean to include this? Can you make TrieUpdate Send and Sync directly?

Reply:
> Do you mean to include this?

Yes, this ensures that no future modifications cause `TrieUpdate` to become `!Send` or `!Sync`.

> Can you make TrieUpdate Send and Sync directly?

Not sure I follow. While I could implement these `unsafe` traits manually, we'd lose compiler's guarantees that the type is indeed `Send + Sync` by construction (as opposed to "by intuition" and/or "by rigorous review".) And it'd be another piece of unsafe code in our code base to tip-toe around.

Reply:
Please brace for an uninformed question:
Can you just do the following? 
```
pub struct TrieUpdate : Send + Sync {
```

Reply:
That isn't a valid syntax ^^. You can do a similar thing with supertrait bounds, that is:

```
trait Foo: Send + Sync {}
```

Which then requires that any implementer of `Foo` implements `Send + Sync`, and we could then implement `Foo` for `TrieUpdate`, achieving the same sort of assert. But I'd say that the assertion is much clearer as written. To be precise the alternative would look like

```
#[allow(dead_code)] // probably necessary as the trait is not used otherwise.
trait AssertSendSync: Send + Sync {}
impl AssertSendSync for TrieUpdate {}
```